tests/FormlyMapper/FormlyField/BirthdayFieldTest.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\DynamicFormBundle\Tests\FormlyMapper\FormlyField;
+
+use Linio\DynamicFormBundle\FormlyMapper\FormlyField\BirthdayField;
+use PHPUnit\Framework\TestCase;
+
+class BirthdayFieldTest extends TestCase
+{
+    /**
+     * @var DateField
+     */
+    protected $formlyField;
+
+    public function basicDataProvider(): array
+    {
+        $config = [
+            'name' => 'borndate',
+            'type' => 'birthday',
+            'options' => [
+                'required' => true,
+                'label' => 'Borndate',
+            ],
+        ];
+
+        $expected = [
+            'key' => 'borndate',
+            'type' => 'input',
+            'templateOptions' => [
+                'required' => true,
+                'label' => 'Borndate',
+                'type' => 'birthday',
+            ],
+        ];
+
+        return [
+            [$config, $expected],
+        ];
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingBirthdayFields(array $fieldConfiguration, array $expected): void
+    {
+        $fieldConfiguration['validation'] = [
+            'Symfony\Component\Validator\Constraints\Regex' => [
+                'pattern' => '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$',
+                'message' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $expected['templateOptions']['pattern'] = '^[0-9]{4}\-[0-9]{2}\-[0-9]{2}$';
+        $expected['validation'] = [
+            'messages' => [
+                'regex' => 'The birthday field must follow the pattern "yyyy-MM-dd"',
+            ],
+        ];
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testIsAddingSpecificYears(array $fieldConfiguration, array $expected): void
+    {
+        $years = [2000, 2001, 2002, 2003, 2004, 2005];
+
+        $fieldConfiguration['options']['years'] = $years;
+
+        $expected['templateOptions']['years'] = $years;
+
+        $this->formlyField->setFieldConfiguration($fieldConfiguration);
+        $actual = $this->formlyField->getFormlyFieldConfiguration();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * @dataProvider basicDataProvider
+     */
+    public function testWithNotNumberSpecificYears(array $fieldConfiguration, array $expected): void
```

Comment:
If you break this test name into a sentence, it doesn't make any sense. A better name would be `testItRemovesInvalidYears` or something of the sort, if that is what the test is actually trying to test. It's a little bit confusing on what the test is trying to accomplish IMO.

Reply:
Yes, absolutely right. I improved this test case by throwing an exception